### PR TITLE
Update to bliss 0.77

### LINKED ISSUE #33010: Update bliss to 0.77
<div id="comment:0"></div>

Comes with a cmake-based build system now (albeit with no install target), so the autotoolization patches shouldn't be needed. Not sure about the other patches included in the fork, but in case any of them is still needed we might consider to keep them here rather than in an outside fork for transparency.

Needs changes in sagelib for the new `canonical_form` and `find_automorphisms` API.

CC:  @mkoeppe @kiwifb

Component: **packages: optional**

Author: **Antonio Rojas, François Bissey**

Branch/Commit: **[u/arojas/bliss-0.77](https://github.com/sagemath/sagetrac-mirror/tree/u/arojas/bliss-0.77) @ [`5fb71e5`](https://github.com/sagemath/sagetrac-mirror/commit/5fb71e501c369e3c74a76184bdf020ced08a9fdc)**

_Issue created by migration from https://trac.sagemath.org/ticket/33010_

### PR DESCRIPTION
**Pull request by antonio-rojas**
Comes with a cmake-based build system now (albeit with no install target), so the autotoolization patches shouldn't be needed. Not sure about the other patches included in the fork, but in case any of them is still needed we might consider to keep them here rather than in an outside fork for transparency.

Needs changes in sagelib for the new `canonical_form` and `find_automorphisms` API.

Fixes: #33010

### CODE REVIEW

**Comment by antonio-rojas** [comment]
@kiwifb I think I've pulled all commits from trac including your cmake changes, please double check.

File: src/sage/graphs/bliss_find_automorphisms.h
Comment (dcoudert) [inline src/sage/graphs/bliss_find_automorphisms.h:1]: Is this file necessary ? can't this be done in a file named `bliss.pxd` ? or should we create a subdirectory bliss to store this file and everything related to bliss to have something clean ?

**Comment by kiwifb** [comment]
Looks great and the bots are happy apart from the linter. I will try to figure what it complains about.

File: src/sage/graphs/bliss.pyx [hunk 1]
```diff
@@ -47,22 +47,22 @@ cdef extern from "bliss/graph.hh" namespace "bliss":
...
 
     cdef cppclass Digraph(AbstractGraph):
         Digraph(const unsigned int)
         void add_edge(const unsigned int, const unsigned int)
-        void find_automorphisms(Stats&, void (*)(void*, unsigned int,
-                                                 const unsigned int*), void*)
-        void change_color(const unsigned int, const unsigned int)
-        const unsigned int* canonical_form(Stats&, void (*)(void*, unsigned int,
-                                                            const unsigned int*), void*)
+        void change_color(const unsigned int, const unsigned int);
```
Comment (kiwifb) [inline src/sage/graphs/bliss.pyx]: The linter is complaining about the semi-column (`;`) at the end of the line.

Comment (kiwifb) [reply src/sage/graphs/bliss.pyx]: I am actually surprised it didn't cause the build to fail. please remove it.

**Review by kiwifb — APPROVED** [review]
Good to go

File: src/sage/graphs/bliss.pyx (hunk 1 above)
Comment (dimpase) [reply src/sage/graphs/bliss.pyx]: that's fixed, as far as I see

**Comment by dimpase** [comment]
all good

**Review by dimpase — APPROVED** [review]

**Comment by dimpase** [comment]
it's a part of the updated interface, needed due to API changes. It's a C++
header file, it's cleaner this way, than trying to do the same in Cython
(which in case of C++ is rather arcane). We can move it to a subdirectory
in sage/graphs.

**Comment by dcoudert** [comment]
A subdirectory as done for cliquer might be better.

**Comment by dimpase** [comment]
> A subdirectory as done for cliquer might be better.

this is now https://github.com/antonio-rojas/sage/pull/1 - to be merged by @antonio-rojas

File: src/sage/graphs/bliss_find_automorphisms.h
Comment (dimpase) [reply src/sage/graphs/bliss_find_automorphisms.h:1]: fixed

**Comment by mkoeppe** [comment]
I had to make this change to get it compile on macOS (with Xcode)
```
diff --git a/src/sage/graphs/bliss.pyx b/src/sage/graphs/bliss.pyx
index 373b99f4d5..d8c50f220e 100644
--- a/src/sage/graphs/bliss.pyx
+++ b/src/sage/graphs/bliss.pyx
@@ -1,4 +1,5 @@
 # distutils: language = c++
+# distutils: extra_compile_args = -std=c++11
 # distutils: libraries = bliss
 # sage_setup: distribution = sagemath-bliss
 
```

**Comment by dimpase** [comment]
> I had to make this change to get it compile on macOS (with Xcode)
> 
> ```
> diff --git a/src/sage/graphs/bliss.pyx b/src/sage/graphs/bliss.pyx
> index 373b99f4d5..d8c50f220e 100644
> --- a/src/sage/graphs/bliss.pyx
> +++ b/src/sage/graphs/bliss.pyx
> @@ -1,4 +1,5 @@
>  # distutils: language = c++
> +# distutils: extra_compile_args = -std=c++11
>  # distutils: libraries = bliss
>  # sage_setup: distribution = sagemath-bliss
>  
> ```

OK, added.
